src/main/java/com/example/alaa/university/controller/SubjectController.java

```diff
@@ -0,0 +1,58 @@
+package com.example.alaa.university.controller;
+
+import com.example.alaa.university.domain.Subject;
+import com.example.alaa.university.domain.Teacher;
+import com.example.alaa.university.service.SubjectService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RequestMapping("/api/subjects")
+@RestController
+public class SubjectController {
+    @Autowired
+    private SubjectService subjectService;
```

Comment:
Feild injection is not recomended. it is better to use Constructor injection. You can look on details explanation here https://www.baeldung.com/java-spring-field-injection-cons